packages/paste-core/components/editable-code-block/src/EditableCodeBlock.tsx

```diff
@@ -6,34 +6,149 @@ import {
   type CodeEditorProps,
   type Editor,
   type Monaco,
+  type IRange,
+  type IMarkdownString,
 } from '@twilio-paste/code-editor-library';
+import {ScreenReaderOnly} from '@twilio-paste/screen-reader-only';
+import {Spinner} from '@twilio-paste/spinner';
+import {StylingGlobals} from '@twilio-paste/styling-library';
+
+const LoadingMessage: React.FC<{i18nLoadingLabel: string}> = ({i18nLoadingLabel}) => (
+  <>
+    <Spinner color="colorTextPrimaryWeak" decorative />
+    <ScreenReaderOnly>{i18nLoadingLabel}</ScreenReaderOnly>
+  </>
+);
+LoadingMessage.displayName = 'LoadingMessage';
 
 export interface EditableCodeBlockProps
-  extends Omit<CodeEditorProps, 'wrapperProps' | 'className' | 'loading' | 'theme'> {
+  extends Omit<CodeEditorProps, 'wrapperProps' | 'className' | 'loading' | 'theme' | 'options'> {
+  children?: never;
+  /*
+   * NOTE: I leave these commented out because I'm not sure whether we want to allow users to override these options.
+   * If we do, we can uncomment these easily in the future since the work is already done.
+   */
+  // monacoOptions?: CodeEditorProps['options'];
   element?: BoxProps['element'];
+  minimap?: boolean;
+  lineNumbers?: 'on' | 'off';
+  folding?: boolean;
+  readOnly?: boolean;
+  indentationGuide?: boolean;
+  showMinimap?: boolean;
+  scrollBeyondLastLine?: boolean;
+  inlineErrorRange?: IRange;
+  inlineErrorHoverMessage?: IMarkdownString | IMarkdownString[] | null;
+  inlineErrorIsWholeLine?: boolean;
+  i18nLoadingLabel?: string;
 }
 
```

Comment:
Some other details I'm noticing now in creating the Figma component:

1. Is it possible to use Twilio Mono?
2. Is it possible to set line height to 20px?